apps/dav/tests/unit/carddav/carddavbackendtest.php

```
@@ -36,22 +41,40 @@ class CardDavBackendTest extends TestCase {
 	/** @var CardDavBackend */
 	private $backend;
 
+	/** @var  Principal | \PHPUnit_Framework_MockObject_MockObject */
+	private $principal;
+
+	/** @var  ILogger | \PHPUnit_Framework_MockObject_MockObject */
+	private $logger;
+
+	/** @var  IDBConnection */
+	private $db;
+
 	const UNIT_TEST_USER = 'carddav-unit-test';
 
 	public function setUp() {
 		parent::setUp();
 
-		$principal = $this->getMockBuilder('OCA\DAV\Connector\Sabre\Principal')
+		$this->principal = $this->getMockBuilder('OCA\DAV\Connector\Sabre\Principal')
 			->disableOriginalConstructor()
 			->setMethods(['getPrincipalByPath'])
 			->getMock();
-		$principal->method('getPrincipalByPath')
+		$this->principal->method('getPrincipalByPath')
 			->willReturn([
 				'uri' => 'principals/best-friend'
 			]);
+		$this->logger = $this->getMock('\OCP\ILogger');
+
+		$this->db = \OC::$server->getDatabaseConnection();
+
+		$this->backend = new CardDavBackend($this->db, $this->principal, $this->logger);
+
+		// start every test with a empty cards_properties and cards table
```

It is, because for some card_properties tables I fill them with dummy values. Try it by yourself, remove it and you will see that tests will fail.